numpy/lib/stride_tricks.py

```diff
@@ -111,6 +112,196 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
 
+def _sliding_window_view_dispatcher(x, window_shape, axis=None, *,
+                                    subok=None, writeable=None):
+    return (x,)
+
+
+@array_function_dispatch(_sliding_window_view_dispatcher, module='numpy')
+def sliding_window_view(x, window_shape, axis=None, *,
+                        subok=False, writeable=False):
+    """
+    Create a sliding window view into the array with the given window shape.
+
+    Creates a sliding window view of the N dimensional array with the given
+    window shape. Window slides across each dimension of the array and extract
+    a subsets of the array at any window position.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to create the sliding window view from.
+
+    window_shape : int or tuple of int
+        Size of window over each axis that takes part in the sliding window.
+        If `axis` is not present, must have same length as the number of input
+        array dimensions. Single integers `i` are treated as if they were the
+        tuple `(i,)`.
+
+    axis : int or tuple of int, optional
+        Axis or axes along which the sliding window is applied.
+        By default, the sliding window is applied to all axes and
+        `window_shape[i]` will refer to axis `i` of `x`.
+        If `axis` is given as a `tuple of int`, `window_shape[i]` will refer to
+        the axis `axis[i]` of `x`.
+        Single integers `i` are treated as if they were the tuple `(i,)`.
+
+    subok : bool, optional
+        If True, sub-classes will be passed-through, otherwise the returned
+        array will be forced to be a base-class array (default).
+
+    writeable : bool, optional
+        When true, allow writing to the returned view. The default is false,
+        as this should be used with caution: the returned view contains the
+        same memory location multiple times, so writing to one location will
+        cause others to change.
+
+    Returns
+    -------
+    view : ndarray
+        Sliding window view of the array. The sliding window dimensions are
+        inserted at the end, and the original dimensions are trimmed as
+        required by the size of the sliding window.
+
+        That is, ``view.shape = x_shape_trimmed + window_shape``, where
+        ``x_shape_trimmed`` is ``x.shape`` with every entry reduced by one less
+        than the corresponding window size.
+
+    See Also
+    --------
+    lib.stride_tricks.as_strided: Create a view into the array with the
+        given shape and strides.
+    broadcast_to: broadcast an array to a given shape.
+
+    Notes
+    -----
+    For some cases there may be more efficient approaches to calculate
+    transformations across multi-dimensional arrays, for instance
+    `scipy.signal.fftconvolve`, where combining the iterating step with the
+    calculation itself while storing partial results can result in significant
+    speedups.
+
+    Examples
+    --------
+    >>> x = np.arange(6)
```

Comment:
Should there be an example showing how the sliding window view can be used say to compute a sliding window mean similar to `pd.rolling`

Reply:
I was wondering if it could be used for an IIR filter, although I think SciPy already has a function for that.

Reply:
I am not knowledgable enough to construct an IIR example, but I will include a simple moving average.

Reply:
Thanks for your comments!

I added a simple moving average example. If you consider this not sufficient in terms of application examples, I am open for other suggestions. Otherwise, I will not consider this a blocker for merging.